modules/apps/portal/portal-log4j-test/src/testIntegration/java/com/liferay/portal/log4j/test/PortalLog4jTest.java

```diff
@@ -491,16 +504,47 @@ private void _testLogOutput(
 
 	private static final String _DATE_FORMAT = "yyyy-MM-dd HH:mm:ss.SSS";
 
+	private static final int _DEFAULT_BUFFER_SIZE = 8192;
+
 	private static final Log _log = LogFactoryUtil.getLog(
 		PortalLog4jTest.class);
 
 	private static final Pattern _datePattern = Pattern.compile(
 		"\\d\\d\\d\\d-\\d\\d-\\d\\d \\d\\d:\\d\\d:\\d\\d.\\d\\d\\d");
 	private static Path _tempLogFileDirPath;
+	private static TestOutputStream _testOutputStream;
 	private static Path _textLogFilePath;
-	private static UnsyncStringWriter _unsyncStringWriter;
+	private static final UnsyncStringWriter _unsyncStringWriter =
+		new UnsyncStringWriter();
 	private static Path _xmlLogFilePath;
 
+	private static class TestOutputStream extends CloseShieldOutputStream {
+
+		public TestOutputStream(OutputStream originalOutputStream) {
+			super(originalOutputStream);
+		}
+
+		@Override
+		public void write(byte[] b) throws IOException {
```

Comment:
@yuhai Please rename `byte[] b` to `byte[] bytes`

Reply:
@yuhai Please simplify the logic :
`String content = new String(b);

_unsyncStringWriter.write(content.toCharArray());`